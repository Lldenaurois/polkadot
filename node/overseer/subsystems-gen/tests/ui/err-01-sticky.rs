#![allow(dead_code)]

use polkadot_procmacro_overseer_subsystems_gen::AllSubsystemsGen;

#[derive(Clone, AllSubsystemsGen)]
struct AllSubsystems<A=(),B=A> {
	a: A,

    #[sticky]
    b: B,
}

fn main() {
	let all = AllSubsystems::<u8,u16> {
		a: 0u8,
		b: 1u16,
	};
	let all = all.replace_a(77u8);
	// does not exist
	let all = all.replace_b(1999u16);
	let _ = all;
}
