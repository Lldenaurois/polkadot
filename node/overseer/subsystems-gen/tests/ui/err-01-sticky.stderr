error[E0599]: no method named `replace_b` found for struct `AllSubsystems<u8, u16>` in the current scope
  --> $DIR/err-01-sticky.rs:20:16
   |
6  | struct AllSubsystems<A=(),B=A> {
   | ------------------------------ method `replace_b` not found for this
...
20 |     let all = all.replace_b(1999u16);
   |                   ^^^^^^^^^ help: there is an associated function with a similar name: `replace_a`
